File: HeraldTest-Android/app/src/main/java/com/example/heraldtest/IllnessStatusCode.java
package com.example.heraldtest;

import java.util.HashMap;

/**
 * Represents an epidemiological status for an individual in relation to a single pathogen/toxin.
 * Includes methods for marshalling to/from an integer
 */
public enum IllnessStatusCode {
    susceptible(1),
    infected(2), // Also same for infestation
    transmittable(3),
    illAndTransmittable(4), // aka has developed serious disease, and can transmit
    illAndNonTransmittable(5), // aka has developed serious disease, and cannot transmit
    recovered(6),
    vaccinated(7),
    immune(8),
    nonTransmittable(9); // E.g. host is deceased or isolated.

    // Integer map pattern

    private int value;
    private static HashMap map = new HashMap<>();

    private IllnessStatusCode(int value) {
        this.value = value;
    }

    static {
        for (IllnessStatusCode illnessStatus : IllnessStatusCode.values()) {
            map.put(illnessStatus.value, illnessStatus);
        }
    }

    public static IllnessStatusCode valueOf(int pageType) {
        return (IllnessStatusCode) map.get(pageType);
    }

    public int getValue() {
        return value;
    }
}

File: HeraldTest-iOS/HeraldTest/IllnessStatusCode.swift
//
//  IllnessStatusCode.swift
//  HeraldTest
//
//

enum IllnessStatusCode : Int, CaseIterable {
    case susceptible = 1
    case infected = 2
    case transmittable = 3
    case illAndTransmittable = 4
    case illAndNonTransmittable = 5
    case recovered = 6
    case vaccinated = 7
    case immune = 8
    case nonTransmittable = 9
    
    var name: String {
        switch self {
        case .susceptible: return "susceptible"
        case .infected: return "infected"
        case .transmittable: return "transmittable"
        case .illAndTransmittable: return "illAndTransmittable"
        case .illAndNonTransmittable: return "illAndNonTransmittable"
        case .recovered: return "recovered"
        case .vaccinated: return "vaccinated"
        case .immune: return "immune"
        case .nonTransmittable: return "nonTransmittable"
        }
    }
    
    class IllnessStatusCode {
        
        private var value:Int
        
        private init(_ val: Int) {
            self.value = val
        }
    }
}


